xmtp_pppp/src/encryption.rs

```diff
@@ -0,0 +1,55 @@
+use xmtp_crypto::encryption::{decrypt, encrypt, hkdf};
+use xmtp_proto::xmtp::message_contents::{
+    ciphertext::{Aes256gcmHkdfsha256, Union},
+    Ciphertext,
+};
+
+const PRIVATE_PREFERENCES_ENCRYPTION_KEY_SALT: &[u8] = b"XMTP_PRIVATE_PREFERENCES_ENCRYPTION";
+
+fn derive_encryption_key(private_key: &[u8]) -> Result<[u8; 32], String> {
```

Comment:
Technically, this step is not necessary. There is already a HKDF as part of `xmtp_crypto`'s `encrypt` function. But it does give us some domain separation built in, and is relatively low cost.